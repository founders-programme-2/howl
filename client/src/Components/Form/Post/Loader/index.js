import React from 'react';
import { css } from '@emotion/core';
import { CircleLoader } from 'react-spinners';

const override = css``;

const Loader = ({ uploadFlag }) => {
  return (
    <div className="sweet-loading">
      <CircleLoader
        css={override}
        sizeUnit="px"
        size={15}
        color=" #c02700"
        loading={uploadFlag}
      />
    </div>
  );
};

export default Loader;
