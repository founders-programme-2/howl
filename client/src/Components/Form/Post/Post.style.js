import styled from 'styled-components';
import responsiveSizes from '../../../constants/responsiveSizes';

const { MDM } = responsiveSizes;

const Message = styled.span`
  color: var(--highlight-color);
  font-weight: bold;
  margin: 10px;
`;

const ErrMsg = styled.p`
  color: red;
  font-weight: bold;
`;

const CntrDiv = styled.div`
  @media (max-width: ${MDM}) {
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
  }
`;

export { Message, ErrMsg, CntrDiv };
