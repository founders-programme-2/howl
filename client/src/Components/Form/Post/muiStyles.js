const styles = () => ({
  textField: {
    marginLeft: '5%',
    marginRight: '5%',
    width: '50%',
    background: 'white',
    boxShadow: '0px 4px 4px rgba(0, 0, 0, 0.25)',
    borderRadius: 4,
    paddingLeft: '2%',
    '@media (max-width:780px)': {
      width: '80%',
    },
  },
  textFieldDetails: {
    width: '90%',
  },
  label: {
    paddingLeft: '2%',
  },
  formControlLabel: {
    marginLeft: '2%',
    display: 'block',
    paddingLeft: '2%',
    fontSize: 14,
  },
  uploadIcon: {
    marginLeft: '2%',
    display: 'block',
    paddingLeft: '2%',
  },
  formControl: {
    width: '100%',
  },
});

export default styles;
