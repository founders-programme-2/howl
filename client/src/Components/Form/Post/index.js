import React, { Fragment } from 'react';
import Attach from './attach.png';
import {
  TextField,
  withStyles,
  FormControlLabel,
  Checkbox,
  FormControl,
} from '../../muIndex';
import { Message, ErrMsg, CntrDiv } from './Post.style';
import styles from './muiStyles';
import Loader from './Loader';

const Post = ({
  classes,
  title,
  details,
  imageCap,
  textChange,
  radio,
  imgPermission,
  checkboxChange,
  handleUploadFile,
  message,
  imgLink,
  uploadFlag,
  imageFlag,
  imageCapErr,
  imgLinkErr,
  detailsErr,
  imgPermissionErr,
  titleErr,
}) => {
  const imageCaptionComponent =
    radio === 'imagePost' || radio === 'both' ? (
      <Fragment>
        {imageCapErr ? (
          <ErrMsg>{imageCapErr}</ErrMsg>
        ) : (
          <p>Please enter a caption for this image (under 100 words):</p>
        )}
        <TextField
          id="imageCap"
          label="Image Caption (required)"
          name="imageCap"
          value={imageCap}
          placeholder="Enter image caption..."
          multiline
          rows="2"
          className={classes.textField}
          margin="normal"
          onChange={textChange('imageCap')}
          InputProps={{
            disableUnderline: true,
          }}
          InputLabelProps={{
            className: classes.label,
          }}
        />
      </Fragment>
    ) : null;

  const imageUploadInput =
    (radio === 'imagePost' || radio === 'both') && imageFlag ? (
      <Fragment>
        {imgLinkErr ? (
          <ErrMsg>{imgLinkErr}</ErrMsg>
        ) : (
          <p>Please upload your image:</p>
        )}

        <label htmlFor="inputImg">
          <input
            id="inputImg"
            label="upload file"
            type="file"
            onChange={handleUploadFile}
            accept="image/*"
            hidden
          />
          <img src={Attach} alt="attach" className={classes.uploadIcon} />
        </label>
      </Fragment>
    ) : null;

  const uploadStatusMsg =
    (radio === 'imagePost' || radio === 'both') && imgLink ? (
      <Message>{message}</Message>
    ) : null;

  const LoaderWithMsg =
    (radio === 'imagePost' || radio === 'both') && uploadFlag ? (
      <Fragment>
        <Loader uploadFlag={uploadFlag} />
        <Message>Uploading...</Message>
      </Fragment>
    ) : null;

  const postDetails =
    radio === 'textPost' || radio === 'both' ? (
      <Fragment>
        {detailsErr ? (
          <ErrMsg>{detailsErr}</ErrMsg>
        ) : (
          <p>
            Please enter your story relating to the Women&#39;s Liberation
            Movement below. We encourage detailed memories but request you have
            a story that is longer than 1000 words, you email directly directly
            to us at fakeemail@howl.co.uk
          </p>
        )}

        <TextField
          id="details"
          label="Details (required)"
          name="details"
          value={details}
          placeholder="Enter your story's details"
          multiline
          fullWidth
          rows="4"
          className={`${classes.textField} ${classes.textFieldDetails}`}
          margin="normal"
          onChange={textChange('details')}
          InputProps={{
            disableUnderline: true,
          }}
          InputLabelProps={{
            className: classes.label,
          }}
        />
      </Fragment>
    ) : null;

  const imgPermissionCheckbox =
    radio === 'imagePost' || radio === 'both' ? (
      <Fragment>
        {imgPermissionErr ? <ErrMsg> {imgPermissionErr}</ErrMsg> : null}

        <FormControlLabel
          className={classes.formControlLabel}
          control={
            <Checkbox
              checked={imgPermission}
              onChange={checkboxChange('imgPermission')}
              value="imgPermission"
            />
          }
          label="I have the rights or permissions to upload this image publicly."
        />
      </Fragment>
    ) : null;
  return (
    <FormControl className={classes.formControl}>
      <CntrDiv>
        {titleErr ? (
          <ErrMsg>{titleErr}</ErrMsg>
        ) : (
          <p>
            What would you like to title your contribution to the archive (under
            10 words)?
          </p>
        )}

        <TextField
          id="title"
          label="Title (required)"
          name="title"
          value={title}
          placeholder="Enter your Title..."
          className={classes.textField}
          margin="normal"
          onChange={textChange('title')}
          InputProps={{
            disableUnderline: true,
          }}
          InputLabelProps={{
            className: classes.label,
          }}
        />
        {imageCaptionComponent}
        {imageUploadInput}
        {uploadStatusMsg}
        {LoaderWithMsg}
        {postDetails}
        {imgPermissionCheckbox}
      </CntrDiv>
    </FormControl>
  );
};

export default withStyles(styles)(Post);
