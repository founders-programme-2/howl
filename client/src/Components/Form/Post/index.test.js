import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import axios from 'axios';
import Post from './index';

test('Natural', () => {
  let state = {};
  const props = {
    title: '',
    details: '',
    imageCap: '',
    imgPermission: false,
    checkboxChange: name => event => {
      this.setState({ [name]: event.target.checked });
    },
    textChange: name => event => {
      this.setState({ [name]: event.target.value });
    },
    handleUploadFile: event => {
      const img = new FormData();
      img.append('uploadedImg', event.target.files[0]);
      this.setState({ uploadFlag: true, imageFlag: false }, () => {
        axios.post('/upload', img).then(({ data }) => {
          if (data.success) {
            state = {
              ...state,
              uploadFlag: false,
              imgLink: data.image,
              message: 'Image uploaded successfully!',
            };
          } else {
            state = {
              ...state,
              uploadFlag: false,
              imageFlag: true,
              imgLink: 'no url was retrieved',
              message: 'Something went wrong, please try again.',
            };
          }
        });
      });
    },
    radio: 'textPost',
    imgLink: 'https://s3.envato.com/files/255314612/pre80.png',
    message: '',
    uploadFlag: false,
    imageFlag: true,
  };
  const post = renderer
    .create(
      <BrowserRouter>
        <Post {...props} />
      </BrowserRouter>
    )
    .toJSON();
  expect(post).toMatchSnapshot();
});
