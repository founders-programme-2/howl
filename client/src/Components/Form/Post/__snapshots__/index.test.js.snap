// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Natural 1`] = `
<div
  className="MuiFormControl-root Post-formControl-6"
>
  <div
    className="sc-htpNat BFnTC"
  >
    <p>
      What would you like to title your contribution to the archive (under 10 words)?
    </p>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root Post-textField-1"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Post-label-3"
        data-shrink={false}
        htmlFor="title"
      >
        Title (required)
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="title"
          name="title"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter your Title..."
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <p>
      Please enter your story relating to the Women's Liberation Movement below. We encourage detailed memories but request you have a story that is longer than 1000 words, you email directly directly to us at fakeemail@howl.co.uk
    </p>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root Post-textField-1 Post-textFieldDetails-2"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Post-label-3"
        data-shrink={false}
        htmlFor="details"
      >
        Details (required)
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
        onClick={[Function]}
      >
        <textarea
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          disabled={false}
          id="details"
          name="details"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter your story's details"
          required={false}
          rows="4"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
